packages/medusa/src/utils/middlewares/ensure-publishable-api-key.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  MedusaNextFunction,
+  MedusaResponse,
+  MedusaStoreRequest,
+} from "@medusajs/framework"
+import {
+  ApiKeyType,
+  isPresent,
+  MedusaError,
+  PUBLISHABLE_KEY_HEADER,
+} from "@medusajs/utils"
+import { refetchEntity } from "../../api/utils/refetch-entity"
+
+export function ensurePublishableApiKey() {
```

Comment:
suggestion: Albeit not something we typically do, here it might make sense to throw a more guiding error message, because it is otherwise keeping them from getting started. Something like:
```
Publishable API key required in the request header: {PUBLISHABLE_KEY_HEADER}.
You can manage your keys in settings in the dashboard.
```

Wdyt?

Reply:
Much better, changed